pkg/kvstore/allocator/allocator.go

```diff
@@ -282,11 +282,11 @@ func NewAllocator(basePath string, typ AllocatorKey, opts ...AllocatorOption) (*
 		return nil, errors.New("Maximum ID must be greater than minimum ID")
 	}
 
-	if err := a.startWatchAndWait(); err != nil {
-		return nil, err
-	}
+	go func() {
+		a.startWatchAndWait()
```

Comment:
Please catch the error and error out with fatal otherwise we may continue without watching for identity changes

Reply:
Done